src/Admin/AbstractAdmin.php

```diff
@@ -2461,9 +2466,11 @@ public function getTranslationDomain()
      *
      * NEXT_MAJOR: remove this method
      *
+     * @param LegacyTranslatorInterface|TranslatorInterface $translator
+     *
      * @deprecated since sonata-project/admin-bundle 3.9, to be removed with 4.0
      */
-    public function setTranslator(TranslatorInterface $translator)
+    public function setTranslator($translator)
```

Comment:
```suggestion
    public function setTranslator(object $translator)
```
?

Reply:
unfortunately not possible.

`setTranslator` is part of this [interface](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AdminInterface.php#L93). We can remove the typehint on `AbstractAdmin but we cannot change it :confused:

Reply:
Actually shall we also change `AdminInterface`? The only way I see is something like this to keep BC

```diff
diff --git a/src/Admin/AdminInterface.php b/src/Admin/AdminInterface.php
index a5205820..dec83ab2 100644
--- a/src/Admin/AdminInterface.php
+++ b/src/Admin/AdminInterface.php
@@ -33,8 +33,9 @@ use Sonata\Form\Validator\ErrorElement;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Translation\TranslatorInterface as LegacyTranslatorInterface;
 use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
@@ -64,6 +65,7 @@ use Symfony\Component\Validator\Validator\ValidatorInterface;
  * @method void                            reorderFormGroup(string $group, array $keys)
  * @method void                            defineFormBuilder(FormBuilderInterface $formBuilder)
  * @method string                          getPagerType()
+ * @method void                            setTranslator($translator);
  */
 interface AdminInterface extends AccessRegistryInterface, FieldDescriptionRegistryInterface, LifecycleHookProviderInterface, MenuBuilderInterface, ParentAdminInterface, UrlGeneratorInterface
 {
@@ -90,10 +92,8 @@ interface AdminInterface extends AccessRegistryInterface, FieldDescriptionRegist
      */
     public function getDatagridBuilder();
 
-    public function setTranslator(TranslatorInterface $translator);
-
     /**
-     * @return TranslatorInterface
+     * @return TranslatorInterface|LegacyTranslatorInterface
      */
     public function getTranslator();

Reply:
It look good for me. Then we can remove LegacyTranslatorInterface, becouse we dont use it. if people use it, then they should test it.